include/swift/SILOptimizer/Utils/InstOptUtils.h

```diff
@@ -587,6 +587,14 @@ FullApplySite cloneFullApplySiteReplacingCallee(FullApplySite applySite,
                                                 SILValue newCallee,
                                                 SILBuilderContext &builderCtx);
 
+/// Analyze the destructor for the class of \p value to see if any instructions
+/// in it could have side effects on the program outside the destructor.
+bool doesDestructorHaveSideEffects(SILValue value);
+
+/// Analyzing the body of this class destructor is valid because the object is
+/// dead. This means that the object is never passed to
+/// objc_setAssociatedObject, so its destructor cannot be extended at runtime.
```

Comment:
This comment is not very good (It was wrong in the original code, too).
It should be something like, "Returns the destructor of \p value ... Returns null if value is not a class or the destructor is not visible or cannot be analyzed"